src/sentry/plugins/base/manager.py

```diff
@@ -248,41 +251,36 @@ def load_handlers(self, cls):
 
         # Registers handlers. Handlers must be in a module directly below
         # the plugin's module:
-
         mod = self.get_plugin_module(cls, 'handlers')
+
         if not mod:
             return
 
-        for _name, impl in inspect.getmembers(mod, inspect.isclass):
-            baseclass = self.get_registered_base_handler(impl)
-            if baseclass and baseclass != impl:
-                # We've found an implementation of the baseclass. Before adding however, we'll
-                # need to make sure that if we already have a less concrete implementation in
-                # the set we should use the more concrete one:
-
-                if baseclass.unique_registration and len(self.handlers[baseclass]) > 1:
-                    # Invariant: There must be only one instance of this baseclass if we're here
-                    assert len(self.handlers[baseclass]) == 1
-
-                    impl_already_reg = self.handlers[baseclass]
-                    if issubclass(impl, impl_already_reg):
-                        # Current class is more concrete, let's replace it:
-                        self.clear_handler_implementations(baseclass)
-                        self.load_handler_implementation(baseclass, impl)
-                    elif issubclass(impl_already_reg, impl):
-                        # The implementation we've already registered is more concrete
-                        pass
-                    else:
-                        # We got two registrations of the same implementation
-                        # TODO: In this case the user should be able to add a config variable
-                        # to select the implementation to use
-                        from clims.handlers import MultipleHandlersNotAllowed
-                        raise MultipleHandlersNotAllowed(
-                            "Trying to register the handler '{}' as the unique implementation "
-                            "of '{}' but already have '{}' registered".format(
-                                impl, baseclass, impl_already_reg))
-                else:
-                    self.handlers[baseclass].add(impl)
+        def find_unique_deepest_impl(clz):
+            sub_classes = clz.__subclasses__()
+            nbr_of_subclasses = len(sub_classes)
+            if nbr_of_subclasses == 0:
+                return clz
+            elif nbr_of_subclasses > 1:
+                raise MultipleHandlersNotAllowed(
+                    "Trying to register the handler '{}' as the unique implementation "
+                    "of '{}' but already have '{}' registered".format(
+                        impl, baseclass, impl_already_reg))
```

Comment:
`impl_already_reg` is not available here. Guess it should now be something like "Trying to ... but found <sub_classes>". Like the solution!

Reply:
That what happens when you copy-paste (and don't write tests).